### PR TITLE
hybrid-structs: add struct for hyrbid reports, build them from existing encrypted orpf reports

### PR DESCRIPTION
**Pull request by eriktaubeneck**

### CODE REVIEW

File: ipa-core/src/report/hybrid.rs [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+pub enum HybridInfo<'a> {
+    Impression(HybridImpressionInfo<'a>),
+    Conversion(HybridConversionInfo<'a>),
+}
+
+impl<BK, V, TS> HybridReport<BK, V, TS>
+where
+    BK: SharedValue,
+    V: SharedValue,
+    TS: SharedValue, // this is only needed for the backpart from EncryptedOprfReport
```
Comment (eriktaubeneck) [inline ipa-core/src/report/hybrid.rs:73]:
```suggestion
    TS: SharedValue, // this is only needed for the backport from EncryptedOprfReport
```

File: ipa-core/src/report/ipa.rs [hunk 2]
```diff
@@ -49,7 +49,7 @@ use crate::{
 };
 
 // TODO(679): This needs to come from configuration.
-static HELPER_ORIGIN: &str = "github.com/private-attribution";
+pub static HELPER_ORIGIN: &str = "github.com/private-attribution";
```
Comment (eriktaubeneck) [inline ipa-core/src/report/ipa.rs:52]:
```suggestion
static HELPER_ORIGIN: &str = "github.com/private-attribution";
```

File: ipa-core/src/report/hybrid.rs [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+        Sum<
+            Sum<<Replicated<BK> as Serializable>::Size, <Replicated<V> as Serializable>::Size>,
+            <Replicated<TS> as Serializable>::Size,
+        >,
+        U16,
+    >: ArrayLength,
+{
+    /// ## Errors
+    /// If the report contents are invalid.
+    pub fn from_bytes<P: PrivateKeyRegistry, B: Deref<Target = [u8]>>(
```
Comment (benjaminsavage) [inline ipa-core/src/report/hybrid.rs:94]: I think you can get rid of the PhantomData if you move the TS bound to this function.

File: ipa-core/src/report/hybrid.rs [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+            .unwrap();
+        let hybrid_report2 =
+            HybridReport::<BA8, BA3, BA20>::from_bytes(enc_report_bytes.as_slice(), &key_registry)
+                .unwrap();
+
+        assert_eq!(hybrid_report, hybrid_report2);
+    }
+
+    #[test]
+    fn convert_to_hybrid_report() {
```
Comment (benjaminsavage) [inline ipa-core/src/report/hybrid.rs:171]: convert_to_hybrid_conversion_report

**Review by akoshelev — APPROVED** [review]

**Review by benjaminsavage — APPROVED** [review]

**Pull request merged**